lib/deploy/stepFunctions/compileIamRole.js

```diff
@@ -429,10 +429,11 @@ function getLambdaPermissions(state) {
     }];
   }
 
-  if (state.Parameters['FunctionName.$']) {
+  if (getParameterOrArgument(state, 'FunctionName.$')) {
```

Comment:
This should also be only `FunctionName `

```suggestion
  if (getParameterOrArgument(state, 'FunctionName')) {
```